cas-server-core-services/src/main/java/org/apereo/cas/services/ServiceRegistryInitializationConfiguration.java

```diff
@@ -0,0 +1,69 @@
+package org.apereo.cas.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.PostConstruct;
+
+
+/**
+ * Initializes Jpa service registry database with available JSON service definitions if necessary (based on configuration flag).
+ *
+ * @author Dmitriy Kopylenko
+ * @author Misagh Moayyed
+ * @since 5.0.0
+ */
+@Configuration("serviceRegistryInitializationConfiguration")
+public class ServiceRegistryInitializationConfiguration {
```

Comment:
Few things:

1. You are never ever supposed to use a Configuration class in your deployment. If you do, fine (and maybe that's the only way you can get X to work for you), maybe you'd like to, but then you'd be on your own. 
2. Classes that are simply marked as public are not necessarily public APIs. Typically, we consider interfaces to be a lot more contractual than other public classes.
3. Even with interfaces, one has to take a pragmatic approach. For instance, we dont really really ever expect anyone to provide their own implementation of CASImpl by implementing that upper interface. (i.e. if you do, you must have a very very good reason or you're doing it wrong) Therefor, pragmatically, neither the impl nor the interface are considered "public APIs". Not saying we should start changing APIs willy nilly, but these are things we have to take into account as well.